apps/meteor/tests/end-to-end/api/09-rooms.js

```diff
@@ -1848,4 +1848,147 @@ describe('[Rooms]', function () {
+				.post(api('rooms.export'))
+				.set(credentials)
+				.send({
+					type: 'file',
+					dateFrom: '2024-03-15',
+					dateTo: '2024-03-22',
+					format: 'html',
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('errorType', 'invalid-params');
+					expect(res.body).to.have.property('error', "must have required property 'rid' [invalid-params]");
+				});
+		});
+
+		it('should fail exporting room as file if no type is provided', async () => {
+			return request
+				.post(api('rooms.export'))
+				.set(credentials)
+				.send({
+					rid: testChannel._id,
+					dateFrom: '2024-03-15',
+					dateTo: '2024-03-22',
+					format: 'html',
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('errorType', 'invalid-params');
+					expect(res.body).to.have.property('error', "must have required property 'type' [invalid-params]");
+				});
+		});
+
+		it('should fail exporting room as file if fromDate is after toDate (incorrect date interval)', async () => {
+			return request
+				.post(api('rooms.export'))
+				.set(credentials)
+				.send({
+					rid: testChannel._id,
+					type: 'file',
+					dateFrom: '2024-03-22',
+					dateTo: '2024-03-15',
+					format: 'html',
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('errorType', 'error-invalid-dates');
+					expect(res.body).to.have.property('error', 'From date cannot be after To date [error-invalid-dates]');
+				});
+		});
+
+		it('should fail exporting room as file if invalid roomId is provided', async () => {
+			return request
+				.post(api('rooms.export'))
+				.set(credentials)
+				.send({
+					rid: 'invalid-rid',
+					type: 'file',
+					dateFrom: '2024-03-22',
+					dateTo: '2024-03-15',
+					format: 'html',
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('errorType', 'error-invalid-room');
+				});
+		});
+
+		it('should fail exporting room as file if an invalid format is provided', async () => {
+			return request
+				.post(api('rooms.export'))
+				.set(credentials)
+				.send({
+					rid: testChannel._id,
+					type: 'file',
+					dateFrom: '2024-03-15',
+					dateTo: '2024-03-22',
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('errorType', 'error-invalid-format');
+				});
+		});
+
+		it('should succesfully export room as file', async () => {
```

Comment:
Ohhh, good catch
I thought of using a delay + our `Triggered_Emails_Count` setting, wdyt? :thinking:
So that we don't need a new flow for TEST_MODE=true

Reply:
I've just added some new end-to-end tests to validate the content of the exported file. Please check this too :hugs: